Source/Core/Core/IOS/ES/NandUtils.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Core/IOS/ES/Formats.h"
+
+namespace IOS
+{
+namespace ES
+{
+TMDReader FindImportTMD(u64 title_id);
+TMDReader FindInstalledTMD(u64 title_id);
+
+// Returns a vector of title IDs in the /title directory, without checking for TMDs at all.
+std::vector<u64> GetInstalledTitles();
+// Returns a vector of title IDs in the /import directory, without checking for TMDs at all.
+std::vector<u64> GetTitleImports();
+// Returns a vector of title IDs for which there is a ticket.
+std::vector<u64> GetTitlesWithTickets();
+
+std::vector<Content> GetStoredContentsFromTMD(const TMDReader& tmd);
```

Comment:
`Content` and `TMDReader` can be forward declared (yes, you can forward-declare types in a `std::vector`. You can also do this with `std::list` and `std::forward_list`, see [standard doc](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4510.html)), allowing you to remove the `Core/IOS/ES/Formats.h` include.